tests/unit/framework/test_pipeline.py

```diff
@@ -216,54 +232,37 @@ def test_enable_iteration_sets_iterate_to_true_stop_to_false(self, _):
         self.pipeline.stop()
         assert not self.pipeline._iterate()
 
-    @mock.patch("logging.Logger.error")
-    def test_critical_input_error_is_logged_and_stored_as_failed(self, mock_error, _):
-        def raise_critical_input_error(event):
-            raise CriticalInputError("An error message", event)
-
-        self.pipeline._setup()
-        self.pipeline._input.get_next.side_effect = raise_critical_input_error
-        self.pipeline._output = {
-            "dummy": original_create({"dummy": {"type": "dummy_output"}}, mock.MagicMock())
-        }
-        self.pipeline.process_pipeline()
-        assert len(self.pipeline._output["dummy"].events) == 0
-        mock_error.assert_called()
-        assert re.search(
-            "A critical error occurred for input .*: An error message", mock_error.call_args[0][0]
-        )
-        assert len(self.pipeline._output["dummy"].failed_events) == 1
-
```

Comment:
Why do you delete this test completely while you rewrote the test for the output connector? Can't you rewrite this one as well?

Reply:
this was tests twice so it is redundant. look for `test_critical_input` and the first tests is the same as this